ADOs/nft-staking/src/execute.rs

```diff
@@ -0,0 +1,802 @@
+use andromeda_std::{
+    ado_contract::ADOContract,
+    common::{context::ExecuteContext, encode_binary, Milliseconds},
+};
+use cosmwasm_std::{coin, ensure, Addr, BankMsg, CosmosMsg, Response, Uint128, WasmMsg};
+use cw721::{Cw721ExecuteMsg, Cw721ReceiveMsg};
+
+use crate::{
+    config::MIN_UNBONDING_PERIOD,
+    query::query_config,
+    state::{
+        get_asset_detail, get_staker_detail, process_pending_rewards, AssetDetail, Config,
+        StakerDetail, ASSET_DETAILS, CONFIG, REWARDS_PER_TOKEN, STAKER_DETAILS,
+    },
+    ContractError,
+};
+
+pub fn receive_cw721(ctx: ExecuteContext, msg: Cw721ReceiveMsg) -> Result<Response, ContractError> {
+    let ExecuteContext {
+        deps, info, env, ..
+    } = ctx;
+    let nft_address = info.sender;
+    let staker = msg.sender;
+    let staker_addr = Addr::unchecked(staker.clone());
```

Comment:
Validating the address would be safer

Reply:
Updated to use validated address intstead of `Addr::unchecked`